src/Raven.Server/Documents/Replication/IncomingReplicationHandler.cs

```diff
@@ -1122,7 +1121,7 @@ public override int Execute(DocumentsOperationContext context)
                                                 AttachmentsStorage.AssertAttachments(document, item.Flags);
 #endif
                                                 database.DocumentsStorage.Put(context, item.Id, null, document, item.LastModifiedTicks, rcvdChangeVector,
-                                                    item.Flags, NonPersistentDocumentFlags.FromReplication);
+                                                    item.Flags.Strip(DocumentFlags.FromClusterTransaction), NonPersistentDocumentFlags.FromReplication);
```

Comment:
Why is this stripped here?

Reply:
You are right.

need to strip it only if `removeClusterTx == true`.